pkg/server/gitproviders/gitprovider.go

```diff
@@ -12,29 +12,39 @@ import (
 
 	"github.com/daytonaio/daytona/cmd/daytona/config"
 	"github.com/daytonaio/daytona/pkg/gitprovider"
+	"github.com/google/uuid"
 )
 
+type GitProviderWithId struct {
```

Comment:
No need for this, instead of `gitProviderList` you can use `gitProviderMap` and have the ID be the key